Shared/Get-ExchangeBuildVersionInformation.ps1

```diff
@@ -4,44 +4,52 @@
 Function Get-ExchangeBuildVersionInformation {
     [CmdletBinding()]
     param(
+        # Parameter help description
+        [Parameter(Mandatory = $true)]
         [object]$AdminDisplayVersion
     )
     begin {
         Write-Verbose "Calling: $($MyInvocation.MyCommand)"
-        Write-Verbose "Passed $($AdminDisplayVersion.ToString())"
-        $AdminDisplayVersion = $AdminDisplayVersion.ToString()
+        Write-Verbose "Passed $([string]$AdminDisplayVersion)"
+        $AdminDisplayVersion = [string]$AdminDisplayVersion
         $exchangeMajorVersion = [string]::Empty
         [int]$major = 0
         [int]$minor = 0
         [int]$build = 0
         [int]$revision = 0
         $product = $null
         [double]$buildVersion = 0.0
+        $oldErrorAction = $ErrorActionPreference
```

Comment:
Cannot have the code written this way inside of `Shared`. 

This will break the logic in place for capturing all the errors that do occur and are counted.